akka-cluster-tools/src/test/java/akka/cluster/client/ClusterClientTest.java

```diff
@@ -57,10 +52,90 @@ public void demonstrateUsage() {
+    private final ActorRef targetClient;
+    private final Set<ActorSelection> contactPoints = new HashSet<>();
+    private boolean isStarted = false;
+
+    public ClientListener(ActorRef targetClient) {
+      this.targetClient = targetClient;
+    }
+
+    public void preStart() {
+      context().system().eventStream().subscribe(self(), ContactPointChange.class);
+      targetClient.tell(GetContactPoints.getInstance(), sender());
+    }
+
+    public void onReceive(Object message) {
+      if (message instanceof ContactPointAdded) {
+        ContactPointAdded msg = (ContactPointAdded) message;
+        if (msg.clusterClient() == targetClient) {
+          contactPoints.add(msg.contactPoint());
+          // if (isStarted) now do something with an up-to-date "contactPoints"
+        }
+      } else if (message instanceof ContactPointRemoved) {
+        ContactPointRemoved msg = (ContactPointRemoved)message;
+        if (msg.clusterClient() == targetClient) {
+          contactPoints.remove(msg.contactPoint());
+          // if (isStarted) now do something with an up-to-date "contactPoints"
+        }
+      } else if (message instanceof ContactPoints) {
+        assert !isStarted;
+        ContactPoints msg = (ContactPoints)message;
+        contactPoints.addAll(msg.getContactPoints());
+        isStarted = true;
+      }
+    }
+  }
+  //#clientEventsListener
+
+  //#receptionistEventsListener
+  static public class ReceptionistListener extends UntypedActor {
+    private final ActorRef targetReceptionist;
+    private final Set<ActorRef> clusterClients = new HashSet<>();
+    private boolean isStarted = false;
+
+    public ReceptionistListener(ActorRef targetReceptionist) {
+      this.targetReceptionist = targetReceptionist;
+    }
+
+    public void preStart() {
+      context().system().eventStream().subscribe(self(), ClusterClientInteraction.class);
+      targetReceptionist.tell(GetClusterClients.getInstance(), sender());
+    }
+
+    public void onReceive(Object message) {
+      if (message instanceof ClusterClientUp) {
+        ClusterClientUp msg = (ClusterClientUp) message;
+        if (msg.clusterReceptionist() == targetReceptionist) {
+          clusterClients.add(msg.clusterClient());
+          // if (isStarted) now do something with an up-to-date "clusterClients"
+        }
+      } else if (message instanceof ClusterClientUnreachable) {
+        ClusterClientUnreachable msg = (ClusterClientUnreachable)message;
+        if (msg.clusterReceptionist() == targetReceptionist) {
+          clusterClients.remove(msg.clusterClient());
+          // if (isStarted) now do something with an up-to-date "clusterClients"
+        }
+      } else if (message instanceof ClusterClients) {
+        assert !isStarted;
+        ClusterClients msg = (ClusterClients)message;
+        clusterClients.addAll(msg.getClusterClients());
+        isStarted = true;
+      }
+    }
+  }
+  //#receptionistEventsListener
```

Comment:
There's not much going on with these Java tests. I ported my Scala code. The test serves mainly to ensure that the Java API is correct.